src/aequitas/squarify_flipped.py

```diff
@@ -0,0 +1,198 @@
+import matplotlib.pyplot as plt
+
+# Attribution: Adapted from squaify source code for plotting treemaps in
+# matplotlib. Treemaps are flipped so smallest square is in bottom right corner.
+# Added squarify_plot_rects function based on source code plot function for
+# plotting predefined rectangles.
+
+def normalize_sizes(sizes, dx, dy):
+    total_size = sum(sizes)
+    total_area = dx * dy
+    sizes = map(float, sizes)
+    sizes = map(lambda size: size * total_area / total_size, sizes)
+    return list(sizes)
+
+
+def layoutrow(sizes, x, y, dx, dy):
+    # generate rects for each size in sizes
+    # dx >= dy
+    # they will fill up height dy, and width will be determined by their area
+    # sizes should be pre-normalized wrt dx * dy (i.e., they should be same units)
+    covered_area = sum(sizes)
+    width = covered_area / dy
+    rects = []
+    heights = 0
+    for size in sizes:
+        rects.append({'x': x, 'y': dy - (size / width), 'dx': width, 'dy': size / width})
+        dy -= size / width
+    return rects
+
+
+def layoutcol(sizes, x, y, dx, dy):
+    # generate rects for each size in sizes
+    # dx < dy
+    # they will fill up width dx, and height will be determined by their area
+    # sizes should be pre-normalized wrt dx * dy (i.e., they should be same units)
+    covered_area = sum(sizes)
+    height = covered_area / dx
+    rects = []
+    for size in sizes:
+        rects.append({'x': x, 'y': dy - height, 'dx': size / height, 'dy': height})
+        x += size / height
+    return rects
+
+
+def layout(sizes, x, y, dx, dy):
+    return layoutrow(sizes, x, y, dx, dy) if dx >= dy else layoutcol(sizes, x, y, dx, dy)
+
+
+def leftoverrow(sizes, x, y, dx, dy):
+    # compute remaining area when dx >= dy
+    covered_area = sum(sizes)
+    width = covered_area / dy
+    leftover_x = x + width
+    leftover_y = dy
+    leftover_dx = dx - width
+    leftover_dy = dy
+    return (leftover_x, leftover_y, leftover_dx, leftover_dy)
+
+
+def leftovercol(sizes, x, y, dx, dy):
+    # compute remaining area when dx < dy
+    covered_area = sum(sizes)
+    height = covered_area / dx
+    leftover_x = x
+    leftover_y = dy
+    leftover_dx = dx
+    leftover_dy = dy - height
+    return (leftover_x, leftover_y, leftover_dx, leftover_dy)
+
+
+def leftover(sizes, x, y, dx, dy):
+    return leftoverrow(sizes, x, y, dx, dy) if dx >= dy else leftovercol(sizes, x, y, dx, dy)
+
+
+def worst_ratio(sizes, x, y, dx, dy):
+    return max([max(rect['dx'] / rect['dy'], rect['dy'] / rect['dx']) for rect in layout(sizes, x, y, dx, dy)])
```

Comment:
no need for the _list_ comprehension -- this constructs a needless data structure, when you're just determining the scalar max:

    return max(
        max(
            rect['dx'] / rect['dy'],
            rect['dy'] / rect['dx']
        )
        for rect in layout(sizes, x, y, dx, dy)
    )